src/witan/cic/tuairisc/charting.clj

```diff
@@ -0,0 +1,158 @@
+
+
+
+(defn line-and-ribbon-plot
+  [{:keys [data
+           chart-title chart-height chart-width
+           x x-title x-format
+           y y-title y-scale y-zero
+           irl iru ir-title
+           orl oru or-title
+           group group-title
+           colors-and-shapes]
+    :or {chart-height full-height
+         chart-width  full-width
+         y-zero       true
+         y-scale      false}}]
+  {:height chart-height
+   :width chart-width
+   :title {:text chart-title
+           :fontSize 24}
+   :config {:legend {:titleFontSize 20
+                     :labelFontSize 14}
+            :axisX {:titleFontSize 16
+                    :labelFontSize 12}
+            :axisY {:titleFontSize 16
+                    :labelFontSize 12}}
+   :data {:values (-> data
+                      (tc/map-columns :ir [irl iru] (fn [lower upper]
+                                                      (format "%,.1f - %,.1f" lower upper)))
+                      (tc/map-columns :or [orl oru] (fn [lower upper]
+                                                      (format "%,.1f - %,.1f" lower upper)))
+                      (tc/rows :as-maps))}
+   :encoding {:y {:scale {:domain y-scale
+                          :zero   y-zero}}}
+   :layer [{:mark "errorband"
+            :encoding {:y {:field iru :title y-title :type "quantitative"}
+                       :y2 {:field irl}
+                       :x {:field x :title x-title :type "temporal" :format x-format}
+                       :color {:field group :title group-title}
+                       :tooltip [{:field group :title group-title}
+                                 {:field x :type "temporal" :format x-format :title x-title}
+                                 {:field y :title y-title}
+                                 {:field :ir :title ir-title}
+                                 {:field :or :title or-title}]}}
+           {:mark "errorband"
+            :encoding {:y {:field oru :title y-title :type "quantitative"}
+                       :y2 {:field orl}
+                       :x {:field x :title x-title :type "temporal" :format x-format}
+                       :color {:field group :title group-title}}}
+           {:mark {:type "line"
+                   :size 5}
+            :encoding {:y {:field y :title y-title :type "quantitative"}
+                       :x {:field x :title x-title :type "temporal" :axis {:format x-format}}
+                       ;; color and shape scale and range must be specified or you get extra things in the legend
+                       :color (color-map data group colors-and-shapes)
+                       :tooltip [{:field group :title group-title}
+                                 {:field x :type "temporal" :format x-format :title x-title}
+                                 {:field y :title y-title}
+                                 {:field :ir :title ir-title}
+                                 {:field :or :title or-title}]}}]})
```

Comment:
Is this (and `line-shape-and-ribbon-plot`) different from adroddiad's versions?

Reply:
this is different from adroddiad as it doesn't have the shapes on it (it just looks too busy). 

Also, they return vega-lite specs rather than charts that have gone through (clerk/vl...) so that I can recombine them into other charts.

I also think they'll evolve differently as the data they need to show is a bit different (not sure if I'm going to do a rule with tooltips or popup shapes on mouseover yet)